crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1672,6 +1690,80 @@ impl KnownInstanceType {
             _ => None,
         }
     }
+
+    fn member(self, db: &'db dyn Db, name: &str) -> Symbol<'db> {
+        match (self, name) {
+            (Self::TypeVar(typevar), "__name__") => Symbol::Type(
+                Type::StringLiteral(StringLiteralType::new(db, typevar.name(db).as_str())),
+                Boundness::Bound,
+            ),
+            (Self::TypeVar(typevar), "__bound__") => Symbol::Type(
+                typevar
+                    .upper_bound(db)
+                    .map(|ty| ty.to_meta_type(db))
+                    .unwrap_or(KnownClass::NoneType.to_instance(db)),
+                Boundness::Bound,
+            ),
+            (Self::TypeVar(typevar), "__constraints__") => Symbol::Type(
+                Type::Tuple(TupleType::new(
+                    db,
+                    typevar
+                        .constraints(db)
+                        .iter()
+                        .map(|ty| ty.to_meta_type(db))
+                        .collect::<Vec<_>>()
+                        .into_boxed_slice(),
```

Comment:
```suggestion
                    typevar
                        .constraints(db)
                        .iter()
                        .map(|ty| ty.to_meta_type(db))
                        .collect(),
```

`FromIter` is implemented for `Box<[T]>` so it should be necessary to go to `Vec` and then `into_boxed_slice` (the implementation probably does but `collect` should hide this for you. You may have to type hint `collect::<Box<_>>`

Reply:
Good call, thanks. (This might make a nice clippy rule?)